### PR TITLE
Bump version to 5.2.0

### PR DESCRIPTION
**Pull request by jasontedor**
This commit bumps the version to 5.2.0, and adds unreleased version
constants for version 5.1.0.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/VersionTests.java [hunk 1]
```diff
@@ -262,18 +270,30 @@ public void testLuceneVersionIsSameOnMinorRelease() {
...
+    private static final Version V_20_0_0_UNRELEASED = new Version(20000099, Version.CURRENT.luceneVersion);
 
     // see comment in Version.java about this test
     public void testUnknownVersions() {
         assertUnknownVersion(V_20_0_0_UNRELEASED);
-        // once we release 5.0.2 and it's added to Version.java we need to remove this constant
+        // once we release 5.0.2 and it's added to Version.java, we need to remove this constant
         assertUnknownVersion(ElasticsearchException.V_5_0_2_UNRELEASED);
+        // once we release 5.1.0 and it's added to Version.java, we need to remove these constants
+        assertUnknownVersion(ElasticsearchException.V_5_1_0_UNRELEASED);
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/VersionTests.java]: I wonder if we'd be better off adding it to Version with the `_UNRELEASED` suffix and making that suffix not come up in tests. There are a lot of these....

**Review by nik9000 — APPROVED** [review]
Can you also bump the versions in the `docs/*/index.asciidoc`?

File: core/src/test/java/org/elasticsearch/VersionTests.java (hunk 1 above)
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/VersionTests.java]: Like, not in this PR, but maybe as a future thing.

Comment (javanna) [inline core/src/test/java/org/elasticsearch/VersionTests.java]: I was wondering too, if this is the way we do it at this point. It feels weird that we have 5.1.0 constant till now, and all of a sudden we have to change all classes that were using it. I understand why we do this in master, not sure about 5.x. What's the reasoning behind this?

Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/VersionTests.java]: I'll address it in a follow-up.

**Comment by jasontedor** [comment]
> Can you also bump the versions in the `docs/*/index.asciidoc`?

@nik9000 I pushed e793feb36f63c4f75f380b6c7c790c14a552936f.

**Comment by jasontedor** [comment]
Thanks for looking @nik9000 and @javanna.

**Pull request merged**